### PR TITLE
Update small grants form

### PR DESCRIPTION
**Pull request by wackerow**
## Description
- Updates small grants form per notion docs / google sheet

Google sheet: https://docs.google.com/spreadsheets/d/1l5t3ar-pbJwzjeNtm-xd7fdGC3EVTWnTzDQ98gOGjn4/edit#gid=0
Notion doc: https://www.notion.so/efdn/Complete-List-of-Web-Form-Edits-for-Website-Team-d131db987707417690ee72fd4a7b3491
Current live form page: https://esp.ethereum.foundation/applicants/small-grants/apply
Preview page: https://deploy-preview-217--ecosystem-support.netlify.app/applicants/small-grants/apply

### CODE REVIEW

**Review by nhsz — APPROVED** [review]
LGTM 🎉

**Comment by wackerow** [comment]
Should be ready for final review

File: src/components/forms/SmallGrantsForm.tsx [hunk 1]
```diff
@@ -1717,6 +1768,77 @@ export const SmallGrantsForm: FC = () => {
...
+              fontSize='input'
+              {...register('referrals', {
+                maxLength: 150
+              })}
+            />
+
+            {errors?.referrals?.type === 'maxLength' && (
+              <Box mt={1}>
+                <PageText as='small' fontSize='helpText' color='red.500'>
+                  Additional info cannot exceed 32768 characters.
```
Comment (nhsz) [inline src/components/forms/SmallGrantsForm.tsx:1836]: Should be `Referrals info cannot exceed 150 characters.`

File: src/components/forms/SmallGrantsForm.tsx [hunk 2]
```diff
@@ -615,6 +623,49 @@ export const SmallGrantsForm: FC = () => {
...
+                  <Box mt={1}>
+                    <PageText as='small' fontSize='helpText' color='red.500'>
+                      Problems being addressed is required.
+                    </PageText>
+                  </Box>
+                )}
+                {errors?.problemBeingSolved?.type === 'maxLength' && (
+                  <Box mt={1}>
+                    <PageText as='small' fontSize='helpText' color='red.500'>
+                      Problems cannot exceed 32768 characters.
```
Comment (nhsz) [inline src/components/forms/SmallGrantsForm.tsx:663]: `Problems being addressed cannot exceed 32768 characters`.

Comment (wackerow) [reply src/components/forms/SmallGrantsForm.tsx:663]:
```suggestion
                      Problems being addressed cannot exceed 32768 characters.
```

File: src/components/forms/SmallGrantsForm.tsx (hunk 1 above)
Comment (wackerow) [reply src/components/forms/SmallGrantsForm.tsx:1836]: Good catch thanks
```suggestion
                  Referrals info cannot exceed 150 characters.
```

**Review by nhsz — APPROVED** [review]

**Pull request merged**